contracts/HealthBlock.sol

```diff
@@ -28,6 +28,67 @@ contract HealthBlock {
         address id;
     }
 
+    event DoctorRequestRaised(address indexed doctor, string indexed doctorName, string credentialsHash);
+    event RequestApproved(address indexed doctor, address indexed provider, uint256 indexed requestId);
+    event RequestRejected(address indexed doctor, address indexed provider, uint256 indexed requestId);
+
+    struct Request {
+        address doctor;
+        string doctorName;
+        string credentialsHash;
+        bool approved;
+    }
+    mapping(address => bool) private isDoctorVerified;
+    mapping(address => Request[]) private doctorRequests;
+
+   /*
+    * @dev Set contract deployer as owner
+    */
+    constructor() {
+        admin = msg.sender; // 'msg.sender' is sender of current call, contract deployer for a constructor
+    }
+
+    modifier onlyAdmin() {
+        require(msg.sender == admin, "Only admin can call this function");
+        _;
+    }
+
+    function verifyDoctor(address doctor) public onlyAdmin {
+        isDoctorVerified[doctor] = true;
+    }
+
+    function raiseRequest(string memory doctorName, string memory credentialsHash) public {
+        require(isDoctorVerified[msg.sender], "Only verified doctors can raise request");
+        Request memory request = Request({
+            doctor: msg.sender,
+            doctorName: doctorName,
+            credentialsHash: credentialsHash,
+            approved: false
+        });
+        doctorRequests[admin].push(request);
+        emit DoctorRequestRaised(msg.sender, doctorName, credentialsHash);
+    }
+
+    function getRequests() public view returns(Request[] memory) {
+        require(msg.sender == admin, "Only admin can call this function");
+        return doctorRequests[admin];
+    }
+
+    function approveRequest(uint256 requestId) public {
+        require(msg.sender == admin, "Only admin can call this function");
+        Request storage request = doctorRequests[admin][requestId];
+        require(!request.approved, "Request is already approved");
+        request.approved = true;
+        emit RequestApproved(request.doctor, admin, requestId);
+    }
+
+    function rejectRequest(uint256 requestId) public {
+        require(msg.sender == admin, "Only admin can call this function");
+        Request storage request = doctorRequests[admin][requestId];
+        require(!request.approved, "Request is already approved");
+        emit RequestRejected(request.doctor, admin, requestId);
```

Comment:
what does this method do? it does not update the status to rejected or remove the request?